src/search.rs

```diff
@@ -92,6 +104,9 @@ impl Search {
     /// Runs the search.
     pub fn search(&self) -> Result<SearchResults> {
         // TODO Return error if search_area or query are None
+        if self.search_area == None || self.query == None {
+            return Err(Box::new(SearchError("Please provide search area and query".into())))
+        }
```

Comment:
Looks great!
One little suggestion though: Could you switch this `if` statement to an `if let`? It's just a little more "Rusty", and I prefer pattern matching on `enum`s over equality comparisons. Example below.
```rust
if let (Some(_), Some(_)) = (self.search_area, self.query) {
    // Run properly
} else {
  // Error
}
```
Also, I think the error should say *how* to provide search area and query, which is by using `Search::new`.
Sorry about being so picky 😆

Reply:
No need to apologize! I'm somewhat new to Rust and this kind of feedback is why I chose to do something in Rust for this year's Hacktoberfest.
I made the changes you suggested :)